### PR TITLE
Move /pkg/kubectl/cmd/{command} to staging

### PR DESCRIPTION
**Pull request by seans3**
* Moves `pkg/kubectl/cmd/{command}` to staging, except for `auth`, `convert`, and `get` which still have kubernetes dependencies.
* This is a very large PR, moving almost half of the `pkg/kubectl` codebase to staging
* Publishes this code to the new kubectl repository: https://github.com/kubernetes/kubectl/pkg/cmd/{command}.
* Code will be available for re-use after publishing.

/kind cleanup
/sig cli
/area kubectl
/area code-organization
/priority important-soon

```release-note
NONE
```

### CODE REVIEW

**Comment by BenTheElder** [comment]
viewing the diff is killing my chrome tab :P

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve

File: staging/src/k8s.io/kubectl/pkg/cmd/profiling.go [hunk 1]
```diff
@@ -30,12 +30,12 @@ var (
...
 )
 
-func addProfilingFlags(flags *pflag.FlagSet) {
+func AddProfilingFlags(flags *pflag.FlagSet) {
 	flags.StringVar(&profileName, "profile", "none", "Name of profile to capture. One of (none|cpu|heap|goroutine|threadcreate|block|mutex)")
 	flags.StringVar(&profileOutput, "profile-output", "profile.pprof", "Name of the file to write the profile to")
 }
 
-func initProfiling() error {
+func InitProfiling() error {
```
Comment (liggitt) [inline staging/src/k8s.io/kubectl/pkg/cmd/profiling.go]: not sure how I feel about exporting this modifying global package vars. would something like this be better?

```go
type ProfileOptions struct {
  ProfileName   string
  ProfileOutput string
}
func (p *ProfileOptions) AddFlags(flags *pflag.FlagSet) {
  ... bind to p.ProfileName, p.ProfileOutput ...
}
func (p *ProfileOptions) Init() error {
  ... switch on p.ProfileName ...
}
func (p *ProfileOptions) func Flush() error {
  ... switch on p.ProfileName, flush to p.ProfileOutput ...
}

Comment (liggitt) [reply staging/src/k8s.io/kubectl/pkg/cmd/profiling.go]: and that options struct could be under `k8s.io/kubectl/pkg/util/profiling`

File: staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_test.go [hunk 2]
```diff
@@ -54,7 +54,7 @@ import (
 )
 
 var (
-	fakeSchema                 = sptest.Fake{Path: filepath.Join("..", "..", "..", "..", "api", "openapi-spec", "swagger.json")}
```
Comment (liggitt) [inline staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_test.go]: was fakeSchema pointing at the real static schema before?

File: staging/src/k8s.io/kubectl/pkg/cmd/delete/delete_test.go [hunk 3]
```diff
@@ -38,9 +38,9 @@ import (
 
 func fakecmd() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:                   "delete ([-f FILENAME] | TYPE [(NAME | -l label | --all)])",
+		Use: "delete ([-f FILENAME] | TYPE [(NAME | -l label | --all)])",
```
Comment (liggitt) [inline staging/src/k8s.io/kubectl/pkg/cmd/delete/delete_test.go]: was gofmt happy with this?

Comment (liggitt) [reply staging/src/k8s.io/kubectl/pkg/cmd/delete/delete_test.go]: ah, no, a later commit undid it

File: staging/src/k8s.io/kubectl/pkg/cmd/profiling.go (hunk 1 above)
Comment (liggitt) [reply staging/src/k8s.io/kubectl/pkg/cmd/profiling.go]: I'll take this as a follow up since this PR is a rebase magnet

**Comment by liggitt** [comment]
/lgtm
/approve

please open a follow up resolving https://github.com/kubernetes/kubernetes/pull/80876#discussion_r309960823

File: staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_test.go (hunk 2 above)
Comment (seans3) [reply staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_test.go]: Yes. There are several instances of this in the tests.

File: staging/src/k8s.io/kubectl/pkg/cmd/profiling.go (hunk 1 above)
Comment (seans3) [reply staging/src/k8s.io/kubectl/pkg/cmd/profiling.go]: I'll follow up on this in the next PR if that is OK.

File: staging/publishing/rules.yaml [hunk 4]
```diff
@@ -643,6 +645,8 @@ rules:
     - repository: client-go
       branch: release-12.0
     - repository: code-generator
+      branch: release-1.15
+    - repository: component-base
```
Comment (nikhita) [inline staging/publishing/rules.yaml]: This doesn't really need to be added to 1.15 anymore. It leads to an extra unneeded sync but is a no-op so I'll clean up redundant rules in a follow-up. Example - https://github.com/kubernetes/kubectl/commit/f16387a692113404c05291b3d41ec50bf51af635

**Pull request merged**